.github/workflows/CoverageCollection.yml

```diff
@@ -54,6 +64,26 @@ jobs:
       - name: ⏬ Checkout repository
         uses: actions/checkout@v2
 
+      - name: 🔁 Extract configurations from pyproject.toml
+        id: getVariables
+        run: |
+          function ReadToml() {
```

Comment:
> Convert to either JavaScript (or Python?) or encapsulate in an Action?

Just add `shell: python` in the step (maybe after `id` and before `run`) and use Python straightaway. See https://github.com/pyTooling/Actions/blob/main/.github/workflows/Parameters.yml#L59-L62.

> It should only run if coveragerc is not given, because then it defaults to pyproject.toml.

Use `if: input.coverage_config == ''`. See https://github.com/pyTooling/Actions/blob/main/.github/workflows/UnitTesting.yml#L74.

> At best, also for coveragerc, we should read the values from there.

Which values do you want/need to read from coveragerc?

Reply:
coveragerc contains the same content:
* output directory of the HTML report.
* output path to the XML report file.

With having them extracted, a user can have a local setup that fits his needs, but CI does run independent and can collect generated outputs without needing `input` variables to double encode these paths.

Reply:
I modified the code on the branch to use Python in the Python "shell". Now I need a dependency installed (`tomli`).  
See https://github.com/pyTooling/pyTooling/runs/4607370478?check_suite_focus=true#step:3:37

The Python shell works in the job even without Python installed via the Python setup action. Any idea how to install a requirement, because both Pythons seam to be isolated.

Reply:
Use one step before the python script:

```
- shell: bash
  run: pip install tomli
```

By default, Python is available on the ubuntu-latest environment. Using the setup-python Action is not necessary. You need to use it only in case you want to explicitly customise the version of Python to be used.

In this case, since setup-python is used later, you might want to reorder the steps:

- First, setup-python.
- Second, pip.
- Third the python script.

Reply:
Fixed in fa10ed07.